pandas/tests/io/excel/test_readers.py

```diff
@@ -164,6 +164,17 @@ def xfail_datetimes_with_pyxlsb(engine, request):
 
 
 class TestReaders:
+    def test_read_excel_type_check(self):
+        # GH 58159
+        df = DataFrame({"bool_column": [True]}, dtype=pd.BooleanDtype.name)
```

Comment:
since `boolean` is a nullable EA, can you also add `None` as a value in this column?